test/test_translator.py

```diff
@@ -138,8 +135,86 @@ def test_parse_polyline() :
         ]
     assert  test_coordinates == valid_coordinates
 
+def test_parse_polyline_null_parameter() :
+
+    test_polyline= None
+    test_coordinates=icone_translator.parse_polyline(test_polyline)
+    expected_coordinates= []
+    assert  test_coordinates == expected_coordinates
+
+def test_parse_polyline_invalid_data() :
+    test_polyline= 'invalid' 
+    test_coordinates=icone_translator.parse_polyline(test_polyline)
+    valid_coordinates= []
+    assert  test_coordinates == valid_coordinates
+
+def test_parse_polyline_invalid_coordinates() :
+    try:
+        test_polyline= 'a,b,c,d' 
+        test_coordinates=icone_translator.parse_polyline(test_polyline)
+        assert False
+    except RuntimeError: 
+        assert True
+
+class TestTranslator(unittest.TestCase):
+  def test_parse_polyline_invalid_coordinates(self):
+    test_polyline = 'a,b,c,d'
+    with pytest.raises(RuntimeError) as runtimeErr:
+        icone_translator.parse_polyline(test_polyline)
+    assert "Failed to parse polyline data." in str(runtimeErr.value)
+
+    
+
+def test_validate_incident_valid_data():
+  test_valid_output = {
+    '@id': 'U13631595_202012160845',
+    'updatetime': '2020-12-16T17:18:00Z',
+    'starttime': '2020-12-07T14:18:00Z',
+    'description': 'Road constructions are going on', 
+    'creationtime': '2020-12-13T14:18:00Z', 
+    'location': {
+      'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650',
+      'street': 'I-70 N'
+    }
+  }
+  assert icone_translator.validate_incident(test_valid_output) == True
+
+def test_validate_incident_missing_required_field_description():
+  test_valid_output = {
+    '@id': 'U13631595_202012160845',
+    'updatetime': '2020-12-16T17:18:00Z',
+    'starttime': '2020-12-07T14:18:00Z',
+    'creationtime': '2020-12-13T14:18:00Z', 
+    'location': {
+      'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650'
+    }
+  }
+  assert icone_translator.validate_incident(test_valid_output) == False
+
+def test_validate_incident_invalid_start_time():
+  test_valid_output = {
+    '@id': 'U13631595_202012160845',
+    'updatetime': '2020-12-16T17:18:00Z',
+    'starttime': '2020-12',
+    'description': 'Road constructions are going on',
+    'creationtime': '2020-12-13T14:18:00Z', 
+    'location': {
+      'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650',
+      'street': 'I-70 N'
+    }
+  }
+  assert icone_translator.validate_incident(test_valid_output) == False
+
+def test_validate_incident_invalid_test_valid_output():
```

Comment:
Let's simplify this name a bit as it's confusing. Maybe `test_validate_incident_invalid`

Reply:
i renamed it to test_validate_incident_invalid